components/concept/storage/src/main/java/mozilla/components/concept/storage/CreditCardsAddressesStorage.kt

```diff
@@ -214,6 +215,44 @@ data class CreditCard(
     }
 }
 
+/**
+ * Credit card autofill entry.
+ *
+ * This contains the data needed to handle autofill but not the data related to the DB record.
+ *
+ * @property guid The unique identifier for this credit card.
+ * @property name The credit card billing name.
+ * @property number The credit card number.
+ * @property expiryMonth The credit card expiry month.
+ * @property expiryYear The credit card expiry year.
+ * @property cardType The credit card network ID.
+ */
+data class CreditCardEntry(
+    val guid: String?,
```

Comment:
Do we even `guid` this for this object? Other than your new check in `validate`, I see only a single usage: in `CreditCardsAdapter#areItemsTheSame`, which could be replaced with a number check.